test/view.js

```diff
@@ -0,0 +1,397 @@
+      var expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId', 'mangoId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+    });
+  });
+
+  describe('Removing item from collection', function() {
+    it('removes item from view as it was included', function() {
+      var model    = (new Model).at('_page');
+  
+      model = addData(model); // Add items to collection
+      model.fn('yellowFruits', function(emit, fruit) {
+        if (fruit.color === 'yellow') {
+          emit(fruit.name + '*' + fruit.color + '*' + fruit.amount*2, '_page.fruits.' + fruit.id);
+        }
+      });
+
+      // Before removing included item
+      var view = model.at('fruits').view('yellowFruits');
+      view.ref('_page.filteredFruits');
+      var expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+      
+      model.del('fruits.bananaId'); // Remove included item
+            
+      // After removing included item
+      expectedFruits = createExpectedResult(model, ['lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+    });
+
+    it('does NOT make any changes to view as item was NOT included', function() {
+      var model    = (new Model).at('_page');
+
+      model = addData(model); // Add items to collection
+      model.fn('yellowFruits', function(emit, fruit) {
+        if (fruit.color === 'yellow') {
+          emit(fruit.name + '*' + fruit.color + '*' + fruit.amount*2, '_page.fruits.' + fruit.id);
+        }
+      });
+
+      // Before removing item NOT included in view
+      var view = model.at('fruits').view('yellowFruits');
+      view.ref('_page.filteredFruits');
+      var expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+
+      model.del('fruits.appleId'); // Remove item
+
+      // After removing item NOT included in view
+      expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+    });    
+  }); 
+
+  describe('Updating item in collection', function() {
+     it('adds updated item to view as conditions ARE now satisfied', function() {
+      var model    = (new Model).at('_page');
+      
+      model = addData(model); // Add items to collection
+      model.fn('yellowFruits', function(emit, fruit) {
+        if (fruit.color === 'yellow') {
+          emit(fruit.name + '*' + fruit.color + '*' + fruit.amount*2, '_page.fruits.' + fruit.id);
+        }
+      });
+
+      // Before updating item
+      var view = model.at('fruits').view('yellowFruits');
+      view.ref('_page.filteredFruits');
+      var expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
```

Comment:
Basically, you are testing two things here. One thing above the comment, which you've basically already tested - thus, you don't need to test it again (it's actually bad from DRY point of view). See line 79 for an exact duplicate of the above test.

The below test on the other hand seems good.